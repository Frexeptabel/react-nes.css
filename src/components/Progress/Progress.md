Progress example:

```js
<Progress value={10} max={60} />
<Progress color="primary" value={20} max={60} />
<Progress color="success" value={30} max={60}/>
<Progress color="warning" value={40} max={60}/>
<Progress color="error" value={50} max={60}/>
<Progress color="pattern"value={60} max={60}/>
```
