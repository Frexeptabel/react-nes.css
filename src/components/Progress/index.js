import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";

const Progress = ({ color, ...props }) => {
  const className = classNames("nes-progress", color && `is-${color}`);
  return <progress className={className} {...props} />;
};

Progress.propTypes = {
  color: PropTypes.oneOf(["primary", "success", "warning", "error", "pattern"])
};

export default Progress;
