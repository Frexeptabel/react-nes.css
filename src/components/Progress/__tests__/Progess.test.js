import React from "react";
import Progress from "../index";
import renderer from "react-test-renderer";

test("The Progress should be rendered properly", () => {
  const component = renderer.create(<Progress value="90" max="100" />);
  expect(component).toMatchSnapshot();
});

test("The Progress should be rendered properly with props", () => {
  const component = renderer.create(
    <Progress color="success" value="90" max="100" />
  );
  expect(component).toMatchSnapshot();
});
