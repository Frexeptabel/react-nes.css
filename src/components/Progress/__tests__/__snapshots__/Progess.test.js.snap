// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Progress should be rendered properly 1`] = `
<progress
  className="nes-progress"
  max="100"
  value="90"
/>
`;

exports[`The Progress should be rendered properly with props 1`] = `
<progress
  className="nes-progress is-success"
  max="100"
  value="90"
/>
`;
